ddtrace/contrib/requests/connection.py

```diff
@@ -72,7 +72,7 @@ def _wrap_send(func, instance, args, kwargs):
 
     cfg = config.get_from(instance)
     service = None
-    if cfg["split_by_domain"] and hostname:
+    if cfg.get("split_by_domain") and hostname:
```

Comment:
@lievan we probably shouldn't need to make this change. Likely we're not copying the config over when we override the pin on the requests session. We should see if we can preserve the config when we override the pin.